src/theory/arith/nl/iand_table.h

```diff
@@ -93,13 +93,41 @@ class IAndTable
    * @param table a function from pairs of integers to integers.
    *        The domain of this function consists of pairs of
    *        integers between 0 (inclusive) and 2^{bitwidth} (exclusive).
+   *        The domain also includes one additional pair (-1, -1), that
+   *        represents the default (most common) value.
    * @return An ite term that represents this table.
    */
   Node createITEFromTable(
       Node x,
       Node y,
       uint64_t granularity,
-      const std::map<std::pair<uint64_t, uint64_t>, uint64_t>& table);
+      const std::map<std::pair<int64_t, int64_t>, uint64_t>& table);
+
+  /**
+   * Returns d_bvandTable[granularity] if it was already computed.
+   * Otherwise, computes it and then returns it.
+   */
+  std::map<std::pair<int64_t, int64_t>, uint64_t> getAndTable(
```

Comment:
```suggestion
  const std::map<std::pair<int64_t, int64_t>, uint64_t>& getAndTable(
```

Reply:
Again, you should be careful with references here, or otherwise you will make copies of large data structures.

Reply:
Fixed by making this function `void` and accessing the updated field directly.